### PR TITLE
feat: upgrade v1.0.x metadata tables

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

part of #774.

### What is changed and how it works?

- add columns for syncer checkpoint
- update column values for syncer checkpoint
- update column values for online DDL meta

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to update the documentation

### CODE REVIEW

File: pkg/binlog/reader/util.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+// GetGTIDsForPos tries to get GTID sets for the specified binlog position (for the corresponding txn).
+// NOTE: this method is very similar with `relay/writer/file_util.go/getTxnPosGTIDs`, unify them if needed later.
+// NOTE: this method is not well tested directly, but more tests have already been done for `relay/writer/file_util.go/getTxnPosGTIDs`.
+func GetGTIDsForPos(ctx context.Context, r Reader, endPos gmysql.Position, parser2 *parser.Parser) (gtid.Set, error) {
+	// start to get and parse binlog event from the beginning of the file.
+	startPos := gmysql.Position{
+		Name: endPos.Name,
+		Pos:  0,
+	}
+	err := r.StartSyncByPos(startPos)
```
Comment (lance6716) [inline pkg/binlog/reader/util.go:41]: defer Close()?

File: pkg/binlog/reader/util.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+				return nil, errors.Errorf("no GTIDs get for position %s", endPos)
+			}
+			var latestGTIDs gtid.Set
+			latestGTIDs, err = gtid.ParserGTID(flavor, latestGSet.String())
+			if err != nil {
+				return nil, terror.Annotatef(err, "parse GTID set %s with flavor %s", latestGSet.String(), flavor)
+			}
+			return latestGTIDs, nil
+		} else if latestPos > endPos.Pos {
+			return nil, errors.Errorf("no GTIDs get for position %s", endPos)
```
Comment (lance6716) [inline pkg/binlog/reader/util.go:132]: not understand, seems error message should be "not a complete GTIDs at position"?

File: pkg/v1dbschema/schema.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	"github.com/pingcap/dm/pkg/terror"
+	"github.com/pingcap/dm/pkg/utils"
+)
+
+// UpdateSchema updates the DB schema from v1.0.x to v2.0.x, including:
+// - update checkpoint.
+// - update online DDL meta.
+func UpdateSchema(tctx *tcontext.Context, db *conn.BaseDB, cfg *config.SubTaskConfig) error {
+	// get db connection.
+	dbConn, err := db.GetBaseConn(tctx.Context())
```
Comment (lance6716) [inline pkg/v1dbschema/schema.go:46]: CloseBaseConn

File: pkg/v1dbschema/schema.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+// - update column value:
+//   - fill `binlog_gtid` based on `binlog_name` and `binlog_pos` if GTID mode enable.
+// NOTE: no need to update the value of `table_info` because DM can get schema automatically from downstream when replicating DML.
+func updateSyncerCheckpoint(tctx *tcontext.Context, dbConn *conn.BaseConn, taskName, tableName, sourceID string, fillGTIDs bool, tcpReader reader.Reader, parser2 *parser.Parser) error {
+	var gs gtid.Set
+	if fillGTIDs {
+		// NOTE: get GTID sets for all (global & tables) binlog position has many problems, at least including:
+		//   - it is a heavy work because it should read binlog events once for each position
+		//   - some binlog file for the position may have already been purge
+		//  so we only get GTID sets for the global position now,
```
Comment (lance6716) [inline pkg/v1dbschema/schema.go:94]:
```suggestion
		// so we only get GTID sets for the global position now,
```

File: pkg/v1dbschema/schema_test.go [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+		t.user = "root"
+	}
+	t.password = os.Getenv("MYSQL_PSWD")
+
+	cfg := config.DBConfig{
+		Host:     t.host,
+		Port:     t.port,
+		User:     t.user,
+		Password: t.password,
+		Session:  map[string]string{"sql_log_bin ": "off"}, // do not enable binlog to break other unit test cases.
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply pkg/v1dbschema/schema_test.go]: done in https://github.com/pingcap/dm/pull/858/commits/00da495d8b4e522ef929c7e0dbf81df8cb74752e.

File: pkg/binlog/reader/util.go (hunk 1 above)
Comment (csuzhangxc) [reply pkg/binlog/reader/util.go:41]: fixed in https://github.com/pingcap/dm/pull/858/commits/00da495d8b4e522ef929c7e0dbf81df8cb74752e.

File: pkg/binlog/reader/util.go (hunk 2 above)
Comment (csuzhangxc) [reply pkg/binlog/reader/util.go]: updated https://github.com/pingcap/dm/pull/858/commits/00da495d8b4e522ef929c7e0dbf81df8cb74752e. this often caused by an invalid position.

File: pkg/v1dbschema/schema.go (hunk 3 above)
Comment (csuzhangxc) [reply pkg/v1dbschema/schema.go:46]: fixed in https://github.com/pingcap/dm/pull/858/commits/00da495d8b4e522ef929c7e0dbf81df8cb74752e.

File: pkg/v1dbschema/schema.go (hunk 4 above)
Comment (csuzhangxc) [reply pkg/v1dbschema/schema.go]: done https://github.com/pingcap/dm/pull/858/commits/00da495d8b4e522ef929c7e0dbf81df8cb74752e.

**Comment by lance6716** [comment]
LGTM

**Pull request merged**